src/normalize/index.js

```diff
@@ -0,0 +1,129 @@
+const { SleepPlacement } = require('../enum')
+const { DEFAULT_OPTIONS } = require('../constants')
+const MIN_SLEEP = 500
+
+/**
+ * Validate archive without iterating nodes
+ * @param {HAR} archive
+ * @return {boolean}
+ */
+function isValidArchive(archive) {
+  return !!archive && !!archive.log && !!archive.log.entries
+}
+
+function getDraft(archive) {
+  const draft =
+    typeof archive === 'string'
+      ? JSON.parse(archive)
+      : JSON.parse(JSON.stringify(archive))
+
+  const { entries = [], ...log } = draft.log
+  draft.log = log
+
+  return [draft, entries]
+}
+
+/**
+ *
+ * @param {Entry[]} nodes
+ * @return {TimelineRef[]}
+ */
+function getTimeline(nodes) {
+  /**
+   * @type {Timeline}
+   */
+  let timeline = []
```

Comment:
An early abort with `!nodes.every(hasStartedDateTime)` alternatively only running if `nodes.every(hasStartedDateTime)` would make more sense I think, thought?

Reply:
Intention was to not iterate entries several times. I guess we could call it "premature optimization".
Will change it to be more straight forward.